lib/active_record/connection_adapters/sqlserver/schema_statements.rb

```diff
@@ -422,10 +422,12 @@ def column_definitions(table_name)
               else
                 type = case ci[:type]
                        when /smallint|int|bigint/ then ci[:_type]
+                       when /bit/ then 'smallint'
```

Comment:
Not sure if that was the best approach to solve that issue. The fact that you had to change a spec rings a bell that something might be wrong there.

TinyTDS casts the `bit` data type to boolean, can't see why we should change it at the adapter level. That could lead to unexpected bugs/changes to already existing apps assuming `boolean` as `true/false` instead of `1/0`.

In any case #875 addressed the original issue already.

Reply:
This change came from #854

Reply:
Right. #854 was closed via #875 and should not be introduced again.